### PR TITLE
[profiler] Expand Kineto platform support

### PR DESCRIPTION
**Pull request by ilia-cher**
Summary:
Expanding support to all builds

Test Plan:
CI

### CODE REVIEW

File: .circleci/scripts/binary_ios_upload.sh [hunk 1]
```diff
@@ -14,7 +14,7 @@ mkdir -p ${ZIP_DIR}/src
 cp -R ${ARTIFACTS_DIR}/arm64/include ${ZIP_DIR}/install/
 # build a FAT bianry
 cd ${ZIP_DIR}/install/lib
-target_libs=(libc10.a libclog.a libcpuinfo.a libeigen_blas.a libpthreadpool.a libpytorch_qnnpack.a libtorch_cpu.a libtorch.a libXNNPACK.a)
+target_libs=(libc10.a libclog.a libcpuinfo.a libeigen_blas.a libpthreadpool.a libpytorch_qnnpack.a libtorch_cpu.a libtorch.a libXNNPACK.a libkineto.a libfmt.a)
```
Comment (malfet) [inline .circleci/scripts/binary_ios_upload.sh:17]: Can you explain why kineto is needed for iOS builds?

File: android/pytorch_android/CMakeLists.txt [hunk 2]
```diff
@@ -117,6 +117,8 @@ if(ANDROID_ABI)
   import_static_lib(libeigen_blas)
   import_static_lib(libcpuinfo)
   import_static_lib(libclog)
+  import_static_lib(libfmt)
```
Comment (malfet) [inline android/pytorch_android/CMakeLists.txt:120]: Hmm, isn't fmt a header only library? (And as such should not generate a static lib)?

File: android/pytorch_android/CMakeLists.txt [hunk 3]
```diff
@@ -117,6 +117,8 @@ if(ANDROID_ABI)
...
+  import_static_lib(libkineto)
```
Comment (malfet) [inline android/pytorch_android/CMakeLists.txt:121]: Can you please elaborate in PR how kineto is used in Android?

File: android/pytorch_android/CMakeLists.txt [hunk 4]
```diff
@@ -134,6 +136,8 @@ if(ANDROID_ABI)
       libeigen_blas
       libcpuinfo
       libclog
+      libfmt
```
Comment (malfet) [inline android/pytorch_android/CMakeLists.txt:139]: Same as above, fmt is header only lib, isn't it?

File: cmake/Dependencies.cmake [hunk 5]
```diff
@@ -1805,6 +1805,7 @@ endif()
 #
 set(TEMP_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
 set(BUILD_SHARED_LIBS OFF CACHE BOOL "Build shared libs" FORCE)
+set(FMT_INSTALL ON CACHE BOOL "")
```
Comment (malfet) [inline cmake/Dependencies.cmake:1808]: Can you please explain why `FMT_INSTALL` is needed?

**Review by malfet — CHANGES_REQUESTED** [review]

File: torch/csrc/autograd/profiler_kineto.h [hunk 6]
```diff
@@ -2,12 +2,9 @@
 
 #include <torch/csrc/autograd/profiler_legacy.h>
 
-// Kineto is currently available on Linux server-side
-#ifdef USE_KINETO
-#if !defined(__linux__) || defined(_WIN32) || defined(C10_MOBILE) || defined(__APPLE__) || defined(DISABLE_KINETO)
+#ifdef DISABLE_KINETO
```
Comment (malfet) [inline torch/csrc/autograd/profiler_kineto.h:5]: Hmm, do you know the history of this change? I.e. why one would define both `USE_KINETO` and `DISABLE_KINETO`?

File: android/pytorch_android/CMakeLists.txt (hunk 2 above)
Comment (ilia-cher) [reply android/pytorch_android/CMakeLists.txt]: right, updated here and in other places

File: android/pytorch_android/CMakeLists.txt (hunk 3 above)
Comment (ilia-cher) [reply android/pytorch_android/CMakeLists.txt:120]: same as in other builds? as in - using `torch.profiler` context manager and saving the trace file

File: .circleci/scripts/binary_ios_upload.sh (hunk 1 above)
Comment (ilia-cher) [reply .circleci/scripts/binary_ios_upload.sh]: using torch.profiler context manager and saving the trace file
removed fmt ref

File: android/pytorch_android/CMakeLists.txt (hunk 4 above)
Comment (ilia-cher) [reply android/pytorch_android/CMakeLists.txt]: updated

File: cmake/Dependencies.cmake (hunk 5 above)
Comment (ilia-cher) [reply cmake/Dependencies.cmake]: removed

File: torch/csrc/autograd/profiler_kineto.h (hunk 6 above)
Comment (ilia-cher) [reply torch/csrc/autograd/profiler_kineto.h:5]: I think that's a debug remnant, will remove

File: android/pytorch_android/CMakeLists.txt (hunk 3 above)
Comment (malfet) [reply android/pytorch_android/CMakeLists.txt:120]: Hmm, but I don't think Android builds have any Python bindings enabled in them, are they?

Comment (ngimel) [reply android/pytorch_android/CMakeLists.txt:120]: Probably not, but going forward we should be able to use profiler with c++ APIs.

Comment (ilia-cher) [reply android/pytorch_android/CMakeLists.txt:120]: sg, let's keep in mind future C++ api, but for now, as discussed, will split the mobile build support in a separate pr

**Comment by ilia-cher** [comment]
updated to exclude android and ios builds

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**